src/utils.js

```diff
@@ -333,6 +331,10 @@ const generateNewCollectionFileName = async ({
 
       newFileName = generateCollectionPageFileName(title, groupIdentifier);
     }
+  } else if (originalThirdNavTitle === thirdNavTitle && collectionPageData) {
+    // Case: renaming/changing details of existing file in collection
+    const groupIdentifier = await generateGroupIdentifier(collectionPageData, thirdNavTitle ? true : false, baseApiUrl, false, thirdNavTitle)
+    newFileName = generateCollectionPageFileName(title, groupIdentifier);
```

Comment:
This fails when trying to rename a collection third nav page to the same third nav section (for example, if I want to rename a page with title `abc` to `def` but to keep it within the same third nav section, `5a-abc.md` should become `5a-def.md`). Currently, it returns `undefined`, or it generates a non-third-nav group identifier by adding it as the last page to the collection (for example, `8-abc.md`)

Reply:
I added the following two cases in the excel sheet https://docs.google.com/spreadsheets/d/1-U69XpMmSHwtOXSYkWGc5gHpwrvA4i23cSQgOA4j5_s/edit#gid=2010487752:

- Rename an existing third nav page while keeping it in the same third nav section
- Rename an existing (normal) collection page while keeping it as a normal collection page

Reply:
Good catch, we should just reuse the existing group identifier in these cases. Resolved in https://github.com/isomerpages/isomercms-frontend/pull/246/commits/4c310d2136e2a1242c4df3e398e65a2c8a98cbba